pkg/plugin/plugin.go

```diff
@@ -200,11 +216,11 @@ func (p *Plugin) Encrypt(ctx context.Context, request *pb.EncryptRequest) (*pb.E
 		case p.healthCheckErrc <- err:
 		default:
 		}
-		zap.L().Error("request to encrypt failed", zap.Error(err))
+		zap.L().Error("request to encrypt failed", zap.String("error-type", ParseError(err).String()), zap.Error(err))
 		failLabel := getStatusLabel(err)
 		kmsLatencyMetric.WithLabelValues(p.keyID, failLabel, operationEncrypt).Observe(getMillisecondsSince(startTime))
 		kmsOperationCounter.WithLabelValues(p.keyID, failLabel, operationEncrypt).Inc()
-		return nil, fmt.Errorf("failed to encrypt data: %v", err)
```

Comment:
Why not wrap with `%w` instead of `%v`?  You can then call [`errrors.Unwrap(err)`](https://pkg.go.dev/errors#Unwrap)  or [`errrors.Is(err)`](https://pkg.go.dev/errors#Is
)